ibis_substrait/expr/operations/relations.py

```diff
@@ -0,0 +1,96 @@
+import typing
+import urllib.parse
+
+import ibis.expr.operations.relations as rel
+import ibis.expr.rules as rlz
+from ibis.common.grounds import Annotable, Comparable
+from ibis.common.validators import validator
+
+
+@validator  # type: ignore[misc]
+def valid_uri(arg: str, **kwargs: typing.Dict[typing.Any, typing.Any]) -> str:
+    arg = rlz.instance_of(str, arg)
+    uri = urllib.parse.urlparse(arg)
+    if not uri.scheme:
+        raise ValueError(f"Missing URI scheme in '{arg}'")
+    return arg
+
+
+class FileFormat(Annotable, Comparable):
```

Comment:
If it is intented to be an ibis operation, then subclass from `ibis.expr.operations.Node`